version: "3.8"
services:

  mongodb:
    image: mongo:6.0
    container_name: challenge_flix_db
    # UTILIZAR EM MAC APPLE SILICON    platform: linux/x86_64
    restart: always
    ports:
      - '27017:27017'
    environment:
      MONGO_INITDB_ROOT_USERNAME: challenge_flix
      MONGO_INITDB_ROOT_PASSWORD: challenge_flix123
      MONGO_INITDB_DATABASE: challenge_flix
    volumes:
      - mongodb_data:/data/db

  aws:
    image: localstack/localstack:latest
    container_name: challenge_flix_aws
    environment:
      - DEBUG=1
      - LOCALSTACK_HOSTNAME=localhost
      - TEST_AWS_ACCOUNT_ID=000000000000
      - EDGE_PORT=4566
      - SERVICES=sqs
    networks:
      - challenge_flix_network
    volumes:
    - ./local/aws:/var/lib/localstack
    - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - 4566:4566


networks:
  challenge_flix_network:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 172.23.0.0/16
          gateway: 172.23.0.1

volumes:
  mongodb_data: